content/articles/cross-site-request-forgery-attack/index.md

```diff
@@ -0,0 +1,146 @@
+
+The tester can use the following methods of testing:
+- Black box testing 
+- Gray box testing
+
+[Black box testing](https://www.javatpoint.com/black-box-testing) is a technique that helps testers find vulnerabilities that make applications and systems exploitable from the outside. It examines the functionality of an application without knowing much about the internal structure.
+
+[Gray box testing](https://www.javatpoint.com/grey-box-testing) is a technique for applications and systems to examine the external and internal workings. It aims to understand and identify errors that belong to applications and systems.
+
+#### Tools for CSRF testing
+1. [CSRF Tester](http://www.owasp.org/index.php/Category:OWASP_CSRFTester_Project)
+2. [Cross Site Requester](http://yehg.net/lab/pr0js/pentest/cross_site_request_forgery.php)
+3. [OWASP ZAP](https://www.zaproxy.org/)
+4. [Pinata CSRF tool](http://code.google.com/p/pinata-csrf-tool/)
+
+### How to prevent CSRF attacks
+IT security experts suggest multiple CSRF prevention techniques. Let's go through some practices to prevent CSRF vulnerabilities:
+- Disable HTTP methods.
+- Make sure the antivirus in your computer is up to date.
+- Do not save passwords and login credentials in the browser.
+- Do not open suspicious emails, navigate to other sites or perform social network communications while authenticated on a web application.
+- Log out entirely from a web application immediately after a session.
+- Carry out regular security tests.
+- Secret token validation.
+- Using referrer headers.
+- Reject unsolicited authentication requests.
+
+For more CSRF prevention alternatives, visit [CSRF prevention cheat sheet](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html) by OWASP.
+
+### Conclusion
+CSRF attack is a severe threat to web applications. The vulnerability depends on how the HTTP protocol manages web requests and processes. In a CSRF attack, the attacker tricks the authenticated user into performing malicious action on a web application without the user's knowledge. This causes a significant impact on the victim or the entire web application.
+
+To summarize, we have looked at:
+- What are CSRF attacks.
+- How CSRF attacks work.
+- Impacts of CSRF attacks.
+- Testing of CSRF attacks.
+- How to prevent CSRF attacks.
+
+I hope this article has given you a better understanding of Cross-Site Request Forgery (CSRF)
+
+### Further reading
+1. [Cross-site request forgery](https://owasp.org/www-community/attacks/csrf)
+2. [Cross-site scripting](https://www.section.io/engineering-education/how-to-prevent-cross-site-scripting-in-node-js/)
+3. [SQL injection](https://www.section.io/engineering-education/how-to-fix-and-prevent-sql-injection-in-wordpress/)
```

Comment:
Please use the actual article titles .ie. 

`How to prevent cross-site scripting in Node.js` instead of `Cross-site scripting`